extensions/azure/processors/PutAzureDataLakeStorage.cpp

```diff
@@ -72,19 +72,26 @@ void PutAzureDataLakeStorage::initialize() {
 }
 
 void PutAzureDataLakeStorage::onSchedule(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSessionFactory>& /*sessionFactory*/) {
-  connection_string_ = getConnectionStringFromControllerService(context);
-  if (connection_string_.empty()) {
+  auto credentials = getCredentialsFromControllerService(context);
+  if (!credentials) {
     throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Azure Storage Credentials Service property missing or invalid");
   }
 
+  if ((!credentials->getUseManagedIdentityCredentials() && credentials->buildConnectionString().empty()) ||
+      (credentials->getUseManagedIdentityCredentials() && credentials->getStorageAccountName().empty())) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Azure Storage Credentials Service properties are not set or invalid");
+  }
+
+  credentials_ = *credentials;
```

Comment:
This onSchedule also sets members but isnt protected by the mutex

Reply:
Same as before.